tests/rules/prefer-concat-latest-from.test.ts

```diff
@@ -1,150 +1,276 @@
+    return this.actions$.pipe(
+      ofType(ProductDetailPage.loaded),
+      concatMap((action) =>
+        of(action).pipe(withLatestFrom(this.store.select(selectProducts))),
+                        ~~~~~~~~~~~~~~ [${messageId}]
+      ),
+      mergeMapTo(of({ type: 'noop' })),
+    )
+  }, { dispatch: false })
+}`,
+        {
+          options: [{ strict: true }],
+          output: stripIndent`
+import { concatLatestFrom } from '@ngrx/effects';
+import { of, withLatestFrom } from 'rxjs'
+@Injectable()
+class Test {
+  effectNOK2 = createEffect(() => ({ debounce = 700 } = {}) => {
+    return this.actions$.pipe(
+      ofType(ProductDetailPage.loaded),
+      concatMap((action) =>
+        of(action).pipe(concatLatestFrom(() => this.store.select(selectProducts))),
+      ),
+      mergeMapTo(of({ type: 'noop' })),
+    )
+  }, { dispatch: false })
+}`,
+        },
+      ),
+      fromFixture(
+        stripIndent`
+import { concatLatestFrom } from '@ngrx/effects'
+import { of, withLatestFrom } from 'rxjs'
+@Injectable()
+class Test {
+  effectNOK3 = createEffect(() => {
+    return condition
+      ? this.actions$.pipe(
+          ofType(ProductDetailPage.loaded),
+          concatMap((action) =>
+            of(action).pipe(
+              withLatestFrom(this.store.select$(something), (one, other) => somethingElse()),
+              ~~~~~~~~~~~~~~ [${messageId}]
+            ),
+          ),
+          mergeMap(([action, products]) => of(products)),
+        )
+      : this.actions$.pipe()
+  })
+}`,
         {
-          messageId: messageId,
-          suggestions: [
-            {
-              messageId: messageIdSuggest,
-              output: stripIndent`
-              import { concatLatestFrom } from '@ngrx/effects';
-              import { of } from 'rxjs';
-              import type { OnRunEffects } from '@ngrx/effects'
-              export class Effect {
-                effect$ = createEffect(
-                  () =>
-                    this.actions$.pipe(
-                      ofType(CollectionApiActions.addBookSuccess),
-                      concatLatestFrom(action => this.store.select(fromBooks.getCollectionBookIds)),
-                      switchMap(([action, bookCollection]) => {
-                        return of({type:'noop'})
-                      }),
-                    ),
-                );
-              }`,
-            },
-          ],
+          options: [{ strict: true }],
+          output: stripIndent`
+import { concatLatestFrom } from '@ngrx/effects'
+import { of, withLatestFrom, map } from 'rxjs'
+@Injectable()
+class Test {
+  effectNOK3 = createEffect(() => {
+    return condition
+      ? this.actions$.pipe(
+          ofType(ProductDetailPage.loaded),
+          concatMap((action) =>
+            of(action).pipe(
+              concatLatestFrom(() => this.store.select$(something),), map( (one, other) => somethingElse()),
+            ),
+          ),
```

Comment:
This is fine, but the ideal scenario would be to also get rid of `concatMap` and `of`.
I'm afraid that this might lead to a lot more code/complexity though.
For now, let's leave it as is.

```suggestion
          concatLatestFrom(() => this.store.select$(something),), map( (one, other) => somethingElse()),
```

Reply:
Yes, I thought about it while implementing and decided it wasn't worth the effort and we could also somehow break something, like for example if someone had structure inside the `concatMap`:

```ts
concatMap((action) => {
  performSideEffects();
  return of(action).pipe(withLatestFrom(...)))
})
```